src/core/playground/signer.ts

```diff
@@ -0,0 +1,58 @@
+import { HexInput } from "../../types";
+import { AccountAddress } from "../accountAddress";
+import { Account } from "./account";
+import { AccountAuthenticatorSingleKey } from "./accountAuthenticator";
+import { Ed25519PrivateKey, type Ed25519PublicKey, type Ed25519Signature } from "./ed25519";
+import type { BaseSigner, PrivateKey, PublicKey } from "./interfaces";
+import { SignatureScheme } from "./scheme";
+import { Secp256k1PrivateKey, type Secp256k1PublicKey, type Secp256k1Signature } from "./secp256k1";
+import { AllowedSignatures, AnyPublicKey, AnySignature } from "./wrapped";
+
+export interface SignerConstructorArgs<
+  TSignature extends AllowedSignatures,
+  TPublicKey extends PublicKey<TSignature> = PublicKey<TSignature>,
+> {
+  privateKey: PrivateKey<TSignature, TPublicKey>;
+  address?: AccountAddress | HexInput;
+}
+
+export interface GenerateSignerArgs<TSignatureScheme extends SignatureScheme = SignatureScheme> {
+  scheme: TSignatureScheme;
+}
+
+export class Signer<
+    TSignature extends AllowedSignatures = AllowedSignatures,
+    TPublicKey extends PublicKey<TSignature> = PublicKey<TSignature>,
+  >
+  extends Account<TSignature, TPublicKey>
+  implements BaseSigner<AccountAuthenticatorSingleKey<TSignature, TPublicKey>>
+{
+  public readonly privateKey: PrivateKey<TSignature, TPublicKey>;
+
+  constructor({ privateKey, address }: SignerConstructorArgs<TSignature, TPublicKey>) {
+    const publicKey = new AnyPublicKey(privateKey.publicKey());
+    super({ publicKey, address });
+    this.privateKey = privateKey;
+  }
+
+  static generate(): Signer<Ed25519Signature, Ed25519PublicKey>;
+  static generate(args: GenerateSignerArgs<SignatureScheme.Ed25519>): Signer<Ed25519Signature, Ed25519PublicKey>;
+  static generate(args: GenerateSignerArgs<SignatureScheme.Secp256k1>): Signer<Secp256k1Signature, Secp256k1PublicKey>;
+  static generate(args?: GenerateSignerArgs) {
+    const scheme = args?.scheme ?? SignatureScheme.Ed25519;
+    if (scheme === SignatureScheme.Ed25519) {
```

Comment:
we should care for legacy vs non-legacy

Reply:
check out my other comment https://github.com/aptos-labs/aptos-ts-sdk/pull/147/files#r1379326061

We needed it to distinguish between the input sdk accepts and the actual enum representation. not sure if with your design is still needed?

Reply:
Yeah I'm fine with removing the re-definition. I was hoping to move away from the "input" sufffix but I see there's a reason behind it.
Removing